engines/producers/app/policies/producers/product_policy.rb

```diff
@@ -0,0 +1,61 @@
+module Producers
+  class ProductPolicy < Shared::ApplicationPolicy
+    def create?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    def show?
+      user_membership_for?(user) || group_membership_for?(user)
+    end
+
+    def update?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    def destroy?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    private
+
+    def user_membership_for?(user)
```

Comment:
Looks like something that we can DRY together with the producer policy. Btw, does pundit impose any limitation on defining the initialize method in the concrete policy classes? I assume not